rfcs/20200612-stream-executor-c-api.md

```diff
@@ -0,0 +1,557 @@
+
+  // Records a start event for an interval timer.
+  TF_BOOL (*start_timer)(
+      SE_Device* executor, SE_Stream stream, SE_Timer timer);
+
+
+  // Records a stop event for an interval timer.
+  TF_BOOL (*stop_timer)(
+      SE_Device* executor, SE_Stream stream, SE_Timer timer);
+
+  /*** MEMCPY CALLBACKS ***/
+  // Entrains a memcpy operation onto stream, with a host destination location
+  // host_dst and a device memory source, with target size size.
+  TF_BOOL (*memcpy_to_host)(
+      SE_Device* executor, SE_Stream stream,
+      void* host_dst,
+      const SE_DeviceMemoryBase* device_src,
+      uint64_t size);
+
+  // Entrains a memcpy operation onto stream, with a device destination location
+  // and a host memory source, with target size size
+
+  TF_BOOL (*memcpy_from_host)(
+      SE_Device* executor, SE_Stream stream,
+      SE_DeviceMemoryBase* device_dst,
+      const void* host_src, uint64_t size);
+
+  // Causes the host code to synchronously wait for operations entrained onto
+  // stream to complete. Effectively a join on the asynchronous device
+  // operations enqueued on the stream before this program point.
+  void (*block_host_until_done)(
+      SE_Device* executor, SE_Stream stream,
+      TF_Status* status);
+
+  // Synchronizes all activity occurring in the StreamExecutor's context (most
+  // likely a whole device).
+  TF_BOOL (*synchronize_all_activity)(SE_Device* executor);
+
+  // Obtains metadata about the underlying device.
+  void (*fill_device_description)(SE_Device* executor,
+                               SE_DeviceDescription* description,
+                               TF_Status* status);
+
+  // Entrains on a stream a user-specified function to be run on the host.
+  TF_BOOL (*host_callback)(SE_Device* executor, SE_Stream* stream,
+                     TF_StatusCallbackFn callback_fn, void* ctx);
+} SE_StreamExecutor;
+
+#define SE_STREAMEXECUTOR_STRUCT_SIZE TF_OFFSET_OF_END(SE_StreamExecutor, host_callback)
+
+typedef struct SE_PlatformParams {
+  size_t struct_size;
+  SE_PlatformId* id;
+  
+  // Callbacks for creating/destroying SE_Device.
+  SE_Device* (*create_device)(SE_Options* options, TF_Status* status);
+  void (*destroy_device)(SE_Device* device);
+  
+  // Callbacks for creating/destroying SE_StreamExecutor.
+  SE_StreamExecutor* (*create_stream_executor)(TF_Status* status);
+  void (*destroy_stream_executor)(SE_StreamExecutor* stream_executor);
+} SE_PlatformParams;
+
+#define SE_PLATFORM_PARAMS_SIZE TF_OFFSET_OF_END(SE_PlatformParams, destroy_stream_executor)
+
+TF_CAPI_EXPORT SE_Platform* SE_NewPlatform(SE_PlatformParams params);
+
+typedef struct SE_PlatformRegistartionParams {
+  size_t struct_size;
+  
+  // Platform name
+  const char* name;
+  size_t name_len;
+  
+  // Device type name. Right now only GPU is supported.
+  char* type;
+  size_t type_len;
+  
+  SE_Platform* platform;
+} SE_PlatformParams;
```

Comment:
Seems its a typo?SE_PlatformParams->SE_PlatformRegistartionParams:)

Reply:
Fixed. Thank you for the catch!